#!/usr/bin/env python
PACKAGE = "rmpcpp_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("r", double_t, 0, "Policy Radius",    1.3, 0.1,   10)
gen.add("c_softmax_obstacle", double_t, 0, "c_softmax_obstacle",   1.0, 0.0001,   10)
gen.add("epsilon_damp", double_t, 0, "epsilon_damp ",    0.1, 0.0001,   10)

gen.add("eta_damp", double_t, 0, "eta_damp",    3.0, 0.0001,   100)
gen.add("v_damp", double_t, 0, "v_damp",    1.0, 0.0001,   100)

gen.add("eta_rep", double_t, 0, "eta_rep",    1.2, 0.0001,   100)
gen.add("v_rep", double_t, 0, "v_rep",    1.4, 0.0001,   100)
gen.add("lin_rep", double_t, 0, "lin_rep",    0.0, 0.0,   100)

gen.add("alpha_goal", double_t, 0, "alpha_goal",    0.8, 0.0001,   100)
gen.add("beta_goal", double_t, 0, "beta_goal",    1.6, 0.0001,   100)
gen.add("gamma_goal", double_t, 0, "beta_goal",    0.35, 0.0001,   1)
gen.add("metric_goal", double_t, 0, "metric_goal",    1,1,   100)
gen.add("metric", bool_t,   0, "directional metric",  True)




exit(gen.generate(PACKAGE, "rmpcpp_planner", "LidarPlanner"))
